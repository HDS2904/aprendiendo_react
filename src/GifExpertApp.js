import React, { useState } from 'react'
import { CategoryAdd } from './components/CategoryAdd';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [ categories, setCategories ] = useState( defaultCategories );

    return (
        <>
            <h2>GifExpertApp</h2>
            <CategoryAdd setCategories={setCategories} />

            <hr/>

            <ul>
                {
                    categories.map( category => 
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    )
                }
            </ul>
        </>
    )
}

export default GifExpertApp;