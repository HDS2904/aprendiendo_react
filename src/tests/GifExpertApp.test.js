import React from 'react'
import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas en el componente GifExpertApp', () => {
  
  test('Evalua si la estructura del componente es igual que la plantilla', () => {
    const wrapper = shallow( <GifExpertApp /> )
    expect( wrapper ).toMatchSnapshot()
  })

  test('Retorna si el numero de categoryas enviada por defauld es la misma', () => {
    const categories = [ 'one piece', 'saint seiya' ]
    const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> )
    expect( wrapper.find('GifGrid').length ).toBe( categories.length )
  })

})
