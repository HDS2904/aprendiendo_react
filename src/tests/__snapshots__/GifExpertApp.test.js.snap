// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Evalua si la estructura del componente es igual que la plantilla 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
